### PR TITLE
add this possible heavy computation in the constructor so it is stored and cached then future usage will be really quick...

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

# Description

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by Dohbedoh** [comment]
We might need to store the value only when the stage / run is finished. A stage may be non restartable when looking at a running pipeline but will be when completed.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java [hunk 1]
```diff
@@ -67,6 +69,18 @@ public PipelineNodeImpl(FlowNodeWrapper node, Reachable parent, WorkflowRun run)
         this.durationInMillis = node.getTiming().getTotalDurationMillis();
         this.self = parent.getLink().rel(node.getId());
         this.parent = parent;
+        // TODO move to use of https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/596
```
Comment (jglick) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java:72]: https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/596

There is an incremental deployment you could pick up (if this were switched to draft).

Comment (olamy) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java:72]: can't really as target branch is `1.25.5.x`

**Comment by olamy** [comment]
> We might need to store the value only when the stage / run is finished. A stage may be non restartable when looking at a running pipeline but will be when completed.

FTR see comment https://github.com/jenkinsci/blueocean-plugin/pull/2415#issuecomment-1471458718

**Comment by olamy** [comment]
mixed with other changes in https://github.com/jenkinsci/blueocean-plugin/pull/2415

**Pull request closed without merging**